aten/src/ATen/cpu/vec256/vec256_complex_double.h

```diff
@@ -0,0 +1,342 @@
+#pragma once
+
+#include <ATen/cpu/vec256/intrinsics.h>
+#include <ATen/cpu/vec256/vec256_base.h>
+#if defined(__AVX__) && !defined(_MSC_VER)
+#include <sleef.h>
+#endif
+
+namespace at {
+namespace vec256 {
+// See Note [Acceptable use of anonymous namespace in header]
+namespace {
+
+#if defined(__AVX__) && !defined(_MSC_VER)
+
+template <> class Vec256<std::complex<double>> {
+private:
+  __m256d values;
+public:
+  using value_type = std::complex<double>;
+  static constexpr int size() {
+    return 2;
+  }
+  Vec256() {}
+  Vec256(__m256d v) : values(v) {}
+  Vec256(std::complex<double> val) {
+    double real_value = std::real(val);
+    double imag_value = std::imag(val);
+    values = _mm256_setr_pd(real_value, imag_value,
+                            real_value, imag_value);
```

Comment:
I'm going to trust that you got the SIMD instructions correct here. If there is some sort of unit test you could write here that would be spiffy.

Reply:
Yes, the unit tests for complex numbers can be found at [pytorch-cpu-strided-complex extension](https://gitlab.com/pytorch-complex/pytorch-cpu-strided-complex).  Once things are running smoothly I can move them into the pytorch unit tests.